roles/openshift_health_checker/openshift_checks/docker_storage_driver.py

```diff
@@ -0,0 +1,50 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class DockerStorageDriver(OpenShiftCheck):
+    """Check Docker storage driver compatibility.
+
+    This check ensures that Docker is using a supported storage driver,
+    and that Loopback is not being used (if using devicemapper).
+    """
+
+    name = "docker_storage_driver"
+    tags = ["preflight"]
+
+    storage_drivers = ["devicemapper", "overlay2"]
+
+    @classmethod
+    def is_active(cls, task_vars):
+        """Skip non-containerized installations."""
+        is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
+        return super(DockerStorageDriver, cls).is_active(task_vars) and is_containerized
+
+    def run(self, tmp, task_vars):
+        info = self.execute_module("docker_info", {}, task_vars).get("info", {})
+
+        if not self.is_supported_storage_driver(info):
+            msg = "Unsupported Docker storage driver detected. Supported storage drivers: {drivers}"
+            return {"failed": True, "msg": msg.format(drivers=self.storage_drivers)}
```

Comment:
This would print something like `...Supported storage drivers: ['foo', 'bar']`.
We can make it nicer with `', '.join(self.storage_drivers)`.